src/equations/shallow_water_multilayer_1d.jl

```diff
@@ -0,0 +1,543 @@
+
+Total energy conservative (mathematical entropy for MLSWE) split form,
+without the hydrostatic pressure.
+When the bottom topography is nonzero this scheme will be well-balanced when used with the 
+nonconservative [`flux_nonconservative_ersing_etal`](@ref).
+
+To obtain an entropy stable formulation the `surface_flux` can be set as
+[`FluxPlusDissipation(flux_ersing_etal, DissipationLocalLaxFriedrichs()), flux_nonconservative_ersing_etal`](@ref).
+
+In the two-layer setting this combination is equivalent to the fluxes in:
+- Patrick Ersing, Andrew R. Winters (2023)
+  An entropy stable discontinuous Galerkin method for the two-layer shallow water equations on 
+  curvilinear meshes
+  [DOI: 10.48550/arXiv.2306.12699](https://doi.org/10.48550/arXiv.2306.12699)
+"""
+@inline function flux_ersing_etal(u_ll, u_rr,
+                                  orientation::Integer,
+                                  equations::ShallowWaterMultiLayerEquations1D)
+    # Unpack left and right state
+    hv_ll = momentum(u_ll, equations)
+    hv_rr = momentum(u_rr, equations)
+
+    # Get the velocities on either side
+    v_ll = velocity(u_ll, equations)
+    v_rr = velocity(u_rr, equations)
+
+    # Initialize flux vector
+    f = zero(MVector{2 * nlayers(equations) + 1, real(equations)})
+
+    # Calculate fluxes in each layer
+    for i in eachlayer(equations)
+        # Compute averages
+        v_avg = 0.5 * (v_ll[i] + v_rr[i])
+        hv_avg = 0.5 * (hv_ll[i] + hv_rr[i])
+
+        f_h = hv_avg
+        # The momentum flux simplifies as the pressure is included in the nonconservative term.
+        f_hv = f_h * v_avg
+
+        setlayer!(f, f_h, f_hv, i, equations)
+    end
+
+    return SVector(f)
+end
+
+# Specialized `DissipationLocalLaxFriedrichs` to avoid spurious dissipation in the bottom
+# topography
+@inline function (dissipation::DissipationLocalLaxFriedrichs)(u_ll, u_rr,
+                                                              orientation_or_normal_direction,
+                                                              equations::ShallowWaterMultiLayerEquations1D)
+    λ = dissipation.max_abs_speed(u_ll, u_rr, orientation_or_normal_direction,
+                                  equations)
+    diss = -0.5 * λ * (u_rr - u_ll)
+    return SVector(@views diss[1:(end - 1)]..., zero(eltype(u_ll)))
+end
+
+@inline function Trixi.max_abs_speeds(u, equations::ShallowWaterMultiLayerEquations1D)
+    h = waterheight(u, equations)
+    hv = momentum(u, equations)
+
+    # Calculate averaged velocity of both layers
+    H = sum(h)
+    v_m = sum(hv) / H
+    c = sqrt(equations.gravity * H)
+
+    return (abs(v_m) + c)
+end
+
+# Calculate approximation for maximum wave speed for local Lax-Friedrichs-type dissipation as the
+# maximum velocity magnitude plus the maximum speed of sound. This function uses approximate
+# eigenvalues as there is no simple way to calculate them analytically.
+
+@inline function Trixi.max_abs_speed_naive(u_ll, u_rr,
```

Comment:
```suggestion
# Calculate approximation for maximum wave speed for local Lax-Friedrichs-type dissipation as the
# maximum velocity magnitude plus the maximum speed of sound. This function uses approximate
# eigenvalues as there is no simple way to calculate them analytically.
@inline function Trixi.max_abs_speed_naive(u_ll, u_rr,
```

Reply:
Done in https://github.com/trixi-framework/TrixiShallowWater.jl/commit/53dfc00487c4527dbad287db70ea222fb6459e43